src/utils/AsyncEventEmitter.ts

```diff
@@ -79,6 +55,16 @@ export class AsyncEventEmitter<
     return super.on(eventName, wrappedListener)
   }
 
+  public once<EventName extends keyof Events>(
+    eventName: EventName,
+    listener: Listener<any>
+  ): this {
+    const wrappedListener = this.wrapListener(eventName, listener)
```

Comment:
`once` listener has to be wrapped the same as the `on` listener to open the event queue once the respective event is emitted.